dpcpp/base/executor.dp.cpp

```diff
@@ -102,6 +102,7 @@ void DpcppExecutor::populate_exec_info(const MachineTopology *mach_topo)
 
 void DpcppExecutor::raw_free(void *ptr) const noexcept
 {
+    queue_->wait_and_throw();
```

Comment:
Just curious here. Isn't this supposed to be `noexcept` ?

Reply:
yeah, I would kind of like to blame SYCL/the DPC++ runtime here, since it frees data before the previous kernel finished running.

Reply:
I guess this is a temporary workaround we shouldn't merge.

Reply:
To be fixed in #832